src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -268,24 +268,37 @@ trait SubmissionComponent {
             val messages: Seq[AttributeString] = record.map {_.messageRecord}.collect { case Some(wm) => AttributeString(wm.message) }
 
             // attach the input resolutions to the workflow object
-            val workflowResolutions = groupedResolutions.get(wr.id).map {seq =>
-              seq.collect {
+            val workflowResolutions: Seq[SubmissionValidationValue] = groupedResolutions.get(wr.id).map { seq =>
+
+              //collect up the workflow resolution results by input
+              val resolutionsByInput = seq.collect {
                 case WorkflowAndInputResolutionRawSqlQuery.WorkflowInputResolutionListResult(workflow, Some(resolution), attribute) =>
-                  // unmarshalAttributes returns a highly nested structure; it is meant for bulk translation of
-                  // attributes, but we use it here for just one
-                  val attr = attribute.map{attrRec =>
-                    submissionAttributeQuery.unmarshalAttributes(Seq( ((attrRec.id,attrRec), None) ))
```

Comment:
Here's the bug. The call to `unmarshalAttributes` is too deep inside the map. At this point we are going through the `WorkflowInputResolutionListResult` records one by one and turning each one into its own input resolution. But if an input resolved to a list, the list will occupy multiple records.